### ISSUE TITLE
Issue with frames

### ISSUE THREAD

**Issue by ssvision** [cambel/ur_ikfast#14]
What are the frames used here for computing Inverse Kinematics ? This for Universal Robots UR5e manipulator
For a joint configuration of [0.849, 0.373, -0.662, 0.782, 1.257, -1.053] I am getting X,Y,Z translation values as 0.410, 0.715,-0.013. 
Ignoring orientation values for now. 
But If i compute the forward kinematics using DH Convention or find the translation values using ros2 tf2 echo commands I get 
-0.410, -0.715,-0.013. 
This is between frame /base_link_inertia and /wrist_3_link. This is the official ROS2 Universal Robotics Description Library.

For the ur5e urdf in this repo i only see links base_link, base, ee_link, tool0. 
Also In the official UR Repo, wrist_3 is connected to flange and flange is connected to tool0. But the urdf in this repo directly connects tool0 to wrist3.
In a nutshell I am utterly confused regarding the starting and ending link of the chain? what is to be used for running it on the real robot? Please Explain.

**Comment by cambel** [comment]
@ssvision 
The frames are `tool0` and `base_link`. That should give you the right values.  Similarly as used for MoveIt:
https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/473cdee66f7840b33efef728c187e1e138ec866e/ur_moveit_config/srdf/ur_macro.srdf.xacro#L14

> Also In the official UR Repo, wrist_3 is connected to flange and flange is connected to tool0. But the urdf in this repo directly connects tool0 to wrist3.

The official URDF was updated a few years back but the position of the frames did not change at all so this library was not updated as it still provides valid results.

**Comment by ssvision** [comment]
Hi @cambel . Thanks a lot for clarifying. The results hold good for /base_link and /wrist_3_link as well for both position as well as orientation which IMHO seems easier to interpret. Thanks a lot for this awesome library. You may close the issue.
